doc/ni_by_example_3.md

```diff
@@ -1,243 +1,356 @@
-#`ni` by Example, Chapter 3 (WIP)
+#`ni` by Example, Chapter 3 (alpha release)
 
 Welcome to the third part of the tutorial. At this point, you've probably activated a certain amount of `ni` productivity by using Hadoop to broadcast your `ni` scripts across hundreds of cores and thousands of jobs.  You should also have a reasonable high-level understanding of how `ni` is a self-modifying quine, and how this allows it to execute across operating systems.
 
-In this chapter, our goal is to multiply your power by introducing more I/O operations, important Perl operations including reducers, cell operations, and ni-specific functions. We'll also demonstrate `ni`'s connections to Python and numpy, Ruby, and Lisp. And we'll show how to patch one of `ni`'s weaknesses (large-scale joins) using `nfu`.
+In this chapter, our goal is to multiply your power by introducing reducers. We'll also discuss some more common I/O operations, give an overview of Perl fundamentals, and demonstrate `ni`'s connections to Python and numpy.
 
-However, we'll first codify `ni` philosophy and style, which we've touched on in the previous two chapters.
+##Perl for `ni`
```

Comment:
I want to say (1) thank you, and (2) you're insane for taking up this cause. Perl advocacy isn't for the faint of heart.